RHEL6_7/init

```diff
@@ -128,10 +128,14 @@ class ModuleInit(object):
 
 if __name__ == "__main__":
 
-    i = ModuleInit(
-        assessment_dir = os.environ['PREUPGM_INIT_ASSESSMENT_DIR'],
-        dst_arch = os.environ['PREUPGM_INIT_DST_ARCH'],
-        contents = os.environ['PREUPGM_INIT_CONTENTS'],
-        scenario = os.environ['PREUPGM_INIT_SCENARIO'],
-    )
+    try:
+        i = ModuleInit(
+            assessment_dir = os.environ['PREUPGM_INIT_ASSESSMENT_DIR'],
+            dst_arch = os.environ['PREUPGM_INIT_DST_ARCH'],
+            contents = os.environ['PREUPGM_INIT_CONTENTS'],
+            scenario = os.environ['PREUPGM_INIT_SCENARIO'],
+        )
+    except KeyError, e:
```

Comment:
This will catch any KeyErrors in `ModuleInit.__init__`.

My suggestion would be to move the try/catch inside the function.  You could achieve that by removing all arguments from the initializer and reading os.environ in there (or passing it as sole argument, eg. `i = ModuleInit(env=os.environ)`.

Reply:
I don't like that solution too much, because it means that you have to call the "exit" function from the `__init__` method and that seems ugly to me. Currently the try/catch is OK according to code inside `__init__` but it doesn't have to be true in future, so I will fix it.

Reply:
I agree with @AloisMahdal and his proposal seems good. Anyway, here's another option, that would get rid of the "ugliness":
```python
if __name__ == "__main__":

    try:
        init_env_vars = {
            'assessment_dir': os.environ['PREUPGM_INIT_ASSESSMENT_DIR'],
            'dst_arch': os.environ['PREUPGM_INIT_DST_ARCH'],
            'contents': os.environ['PREUPGM_INIT_CONTENTS'],
            'scenario': os.environ['PREUPGM_INIT_SCENARIO'],
        }
    except KeyError, e:
        sys.stderr.write("The '%s' environment variable is not set." % str(e))
        sys.exit(1)

    i = ModuleInit(init_env_vars)
```

Reply:
Sorry for the confusion, but I thought this through and nothing seems to really work. and IMO that means this is a bad idea in the first place.  You can't do any useful and maintainable error checking here, because there's no way for lower level (`init`) to babysit upper level (`run_init()`).

IOW, what I said in https://github.com/upgrades-migrations/preupgrade-assistant/pull/311#discussion_r140375004 apples also to this situation.